R/primerTree.R

```diff
@@ -264,3 +267,118 @@ labeled_quantile = function(x, labels, ...){
   names(res) = labels
   res
 }
+
+#' @title calc_rank_dist_ave
+#' Summarize 
+#' pairwise differences.
+
+#' @param x a primerTree object 
+#' @param ... Ignored options
+#' @param ranks ranks to show unique counts for, defaults to the common ranks
+#' @return returns a data frame of results
+#' @details 
+#' The purpose of this function is to calculate the average number
+#' of nucleotide differences between species within a given taxonomic 
+#' level. 
+#' 
+#' There are several key assumptions and calculations made in this 
+#' function.
+#' 
+#' First, the function randomly selects one sequence from each species
+#' in the primerTree results. This is to keep any one species (e.g.
+#' human, cow, etc.) with many hits from skewing the results.
+#' 
+#' Second, for each taxonomic level tested, the function divides the
+#' sequences by each taxon at that level and calculates the mean 
+#' number of nucleotide differences within that taxa, then returns the
+#' mean of those values. 
+#' 
+#' Third, when calculating the average distance, any taxa for which 
+#' there is only one species is omitted, as the number of nucleotide
+#' differences will always be 0.
+#' 
+#' @examples
+#' \dontrun{
+#' calc_rank_dist_ave(mammals_16S)
+#' 
+#' calc_rank_dist_ave(bryophytes_trnL)
+#' 
+#' # Note that the differences between the results from these two primers
+#' # the mean nucleotide differences is much higher for the mammal primers
+#' # than the byrophyte primers. This suggests that the mammal primers have
+#' # better resolution to distinguish individual species.
+#' }
+#' @export
+
+# using tree data format info from http://www.phytools.org/eqg/Exercise_3.2/ 
+calc_rank_dist_ave <- function(x, ranks = common_ranks, ...) {
+  usedRanks <- grep("species", ranks, invert = T, value = T)
```

Comment:
I use snake_case in primerTree (package names cannot have `_` in them).

Also try to follow http://adv-r.had.co.nz/Style.html with the code. Particularly pay attention to whitespace throughout this function.